tests/model_find_all_test.js

```diff
@@ -0,0 +1,328 @@
+        expect(_binaryRetrieved.id.toString()).to.be.equal(buf2.toString());
+      });
+      it('should be able to handle binary values through associations as well...', async function() {
+        const User = this.User;
+        const Binary = this.sequelize.define('Binary', {
+          id: {
+            type: DataTypes.STRING(16, true),
+            primaryKey: true
+          }
+        });
+
+        const buf1 = this.buf;
+        const buf2 = Buffer.alloc(16);
+        buf2.fill('\x02');
+
+        User.belongsTo(Binary, { foreignKey: 'binary' });
+
+        await this.sequelize.sync({ force: true });
+
+        const users = await User.bulkCreate([
+          { username: 'boo5', aBool: false },
+          { username: 'boo6', aBool: true }
+        ]);
+
+        await Binary.bulkCreate([
+          { id: buf1 },
+          { id: buf2 }
+        ]);
+
+        const user = await User.findByPk(users[0].id);
+        const binary = await Binary.findByPk(buf1);
+        await user.setBinary(binary);
+        const _user = await User.findByPk(users[1].id);
+        const _binary = await Binary.findByPk(buf2);
+        await _user.setBinary(_binary);
+        const _binaryRetrieved = await _user.getBinary();
+        const binaryRetrieved = await user.getBinary();
+        expect(binaryRetrieved.id).to.have.length(16);
+        expect(_binaryRetrieved.id).to.have.length(16);
+        expect(binaryRetrieved.id.toString()).to.be.equal(buf1.toString());
+        expect(_binaryRetrieved.id.toString()).to.be.equal(buf2.toString());
+      });
+
+      // Reason: CockroachDB does not yet support CITEXT
+      // Seen here: https://github.com/cockroachdb/cockroach/issues/22463
+      it.skip('should be able to find multiple users with case-insensitive on CITEXT type', async function() {
+        const User = this.sequelize.define('UsersWithCaseInsensitiveName', {
+          username: Sequelize.CITEXT
+        });
+
+        await User.sync({ force: true });
+
+        await User.bulkCreate([
+          { username: 'lowercase' },
+          { username: 'UPPERCASE' },
+          { username: 'MIXEDcase' }
+        ]);
+
+        const users = await User.findAll({
+          where: { username: ['LOWERCASE', 'uppercase', 'mixedCase'] },
+          order: [['id', 'ASC']]
+        });
+
+        expect(users[0].username).to.equal('lowercase');
+        expect(users[1].username).to.equal('UPPERCASE');
+        expect(users[2].username).to.equal('MIXEDcase');
+      });
+    });
+
+    describe('normal findAll', () => {
+      beforeEach(async function() {
+        const user = await this.User.create({ username: 'user', data: 'foobar', theDate: moment().toDate() });
+        const user2 = await this.User.create({ username: 'user2', data: 'bar', theDate: moment().toDate() });
+        this.users = [user].concat(user2);
+      });
+
+      // Reason: The way we treat BigInt ids as Strings, makes the .below and .above comparisons not,
```

Comment:
nit: don't need comma after not. 

I would also reword this as `Because we treat BigInt as Strings, below and above comparisons do not work`